src/Enum.php

```diff
@@ -291,6 +291,8 @@ protected static function resolveFromStaticMethods(ReflectionClass $reflection):
         foreach ($reflection->getMethods(ReflectionMethod::IS_STATIC | ReflectionMethod::IS_PUBLIC) as $method) {
             if (
                 $method->getDeclaringClass()->getName() === self::class
+                || ! $method->isPublic()
```

Comment:
It's a very subjective thing and I would accept this PR but for me I would like it more if it's written like:
```php
|| ! ($method->isPublic() && $method->isStatic())
```

Could be that it's me but I needed a bit to understand it in my head and find out that it's correct.^^
The `or it's not public and static` is a bit more to what we would write it in native language.

@brendt or yourself can decide.

Reply:
That's what I did in the first place but then decided to expand the expression 🙈 Any way is fine for me 🙃

Reply:
I know that @spatie has a standard related to "complex" conditions that should be put into an extra method. So we could introduce a `isPublicAndStatic()` method and/or `isEnumValueMethod()` (just an idea) which contains the whole condition.

I think that an `if` with 4 lines is something like complex.^^